### DISCUSSION TITLE
channel subscription on array contains value?

### DISCUSSION THREAD

**Discussion by pierroo** [supabase/supabase#20307] [Questions]
Hi,

I am trying to subscribe to a channel based on a row containing my user uid.

I have a chat_room table
a "participants" as text[],
and would like to subscribe when a chat_room is created with my uid (text) as part of participants[]

how could I do that?
I tried with this but doesn't trigger:

channelChatRoomInsertRef.current = supabase
      .channel(CHANNEL_CHATROOM_INSERT)
      .on(
        'postgres_changes',
        {
          event: 'INSERT',
          schema: 'public',
          table: 'chat_room',
          filter: `participants=in.(${myUserId})`,
        },

**Comment by GaryAustin1** [comment]
.in() only works with a single value column and will return results for anything in myUserId as an ARRAY.
There is not contains option for realtime filters.
I use RLS to accomplish this as that will limit the payloads to those the user has access to by the RLS policy.

**Reply by pierroo** [reply]
Thank you @GaryAustin1 ; so you are listening to ALL chat_room insert in your example, and then through the RLS it will only return based on your user id for example?
The issue is that I am not using supabase auth, but firebase auth (old app); so I can't rely on the uid decrypted from supabase server in RLS,
and having RLS rely on a uid sent from frontend seems very risky.

Can you enlighten me further on what your RLS would look like exactly?
and if you are indeed listening to all chat_rooms in this example?

**Reply by GaryAustin1** [reply]
I am out most of the day. But I monitor the message table for everything but my RLS checks the chatroom table to see if the user belongs to that chatroom with a join in the RLS.  Your RLS would need to check your array.  You can test your RLS by doing a select on the table you want to monitor and you should only see your own rows.

**Reply by pierroo** [reply]
Interesting, thank you once again @GaryAustin1 
So listening for everything in that message table won't have any impact on the "realtime messages being spread/received" (ie: pricing) even if they are filtered by RLS?
they won't be acknowledged anywhere in the pricing since they are blocked from the RLS directly?
But does it mean every single message added in the table will also trigger RLS for each new message * each users being subscribed to it?

although it may work, I do wonder on the database bandwidth/cost/etc impact it may have?

**Reply by GaryAustin1** [reply]
There will be not impact on cost really as for realtime you get billed by messages and their won't be any if the user can't see the rows.
You really want to have RLS on your table so the user can only see what they are allowed to.   Even if you could filter on realtime the way you want that would not be secure as the user could change the filter.

**Reply by pierroo** [reply]
Very reassuring regarding pricing and bandwidth, thank you.

as for having RLS on table, I just created a separate discussion with my ongoing issue with RLS without being authenticated, one step at a time I'm getting closer :)
https://github.com/orgs/supabase/discussions/20311
